### PR TITLE
Update charming-actions to 2.2.2

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**

### CODE REVIEW

**Comment by taurus-forever** [comment]
> Not sure how to test this PR since it requires a Charmhub secret. @taurus-forever thoughts?

Merge it here due to the PR simplicity and fix quickly if failing, like attempts 1/2 [here](https://github.com/canonical/mysql-k8s-operator/actions/runs/3997519835/jobs/6860548817). Otherwise I can upload secret to some playgound.git

BTW, are you sure 2.2.2 will publish both focal & jammy? I do not recall such fixes there... let's give it a try!

**Comment by taurus-forever** [comment]
> The CI on this PR has failed 5 times—should I disable the failing tests or merge with failing tests or keep retrying?

@shayancanonical what do you know about the tests instabilities here?

**Comment by shayancanonical** [comment]
The failing tests (both `shared-db` and `db-router`) integrate with an external component (keystone). It could be possible that there is an issue in the integration with keystone leading to instabilities. We've seen this in the past

**Review by shayancanonical — COMMENTED** [review]
I believe that the change will allow 2 charm files to be built (in LXD containers instead of on the host machine). However, the `upload_charm` functionality [here](https://github.com/canonical/charming-actions/blob/d60fc89ad72508b04e5cbe4f15f4a6bf11f4eabc/src/services/charmcraft/charmcraft.ts#L191-L207) still expects `charmcraft pack` to produce one charm file

**Comment by shayancanonical** [comment]
my opinion: we should build charms from scratch in a fresh environment when we are building to release them to charmhub

**Comment by taurus-forever** [comment]
> ... still expects charmcraft pack to produce one charm file

Requested improvement: https://github.com/canonical/charming-actions/issues/85

> It could be possible that there is an issue in the integration with keystone leading to instabilities.

keystone is not stable in my local builds as well. Searching a contact for keystone developers to discuss the stability topic.
@carlcsaposs-canonical please invest 2 hours trying to stabilize tests, otherwise temporary disable them (create a Jira ticket to pickup the next sprint, top of backlog please). Tnx!

P.S. [updating](https://github.com/canonical/mysql-operator/pull/97) ops == 2.0.0, just to be up-2-date (it looks mode stable as 1.5.x).

**Comment by taurus-forever** [comment]
> > my opinion: we should build charms from scratch in a fresh environment when we are building to release them to charmhub
> 
> @taurus-forever do you agree? If so, I'll close this PR and open a ticket to release the *.charm files from the build step added in #78

Yes, I agree. Using cache in "building to release" could lead to surprises and we should build in clean env.

**Review by taurus-forever — APPROVED** [review]

**Comment by taurus-forever** [comment]
> Should we wait for [canonical/charming-actions#85](https://github.com/canonical/charming-actions/issues/85) before merging this PR?

No. 99.9% it will take forever.

**Comment by taurus-forever** [comment]
> > > Should we wait for [canonical/charming-actions#85](https://github.com/canonical/charming-actions/issues/85) before merging this PR?
> > 
> > 
> > No. 99.9% it will take forever.
> 
> Why build charms on `focal` and `jammy` if we're only releasing on one series? Furthermore, it seems like it might break the release, or we might accidentally release only `focal` and not `jammy`

In general we need a solution to build and release several series in parallel (e.g. subordinate charm).
Today `focal` can be stop building here (and maybe even removed form charmhub.io) while it will be impossible after the charm production usage. So CI/CD should be designed/optimized for such a requirements.
DPE has to support series until Canonical supports OS: 20.04 => till April 2027 (or even 2032, SOSS).

**Comment by taurus-forever** [comment]
> What I'm confused about is why we should merge this PR before https://github.com/canonical/charming-actions/issues/85 is merged.

Because there is no PR, it is just a issue reported there. Like 24 another issues will stay there until we have a developer to start resolving them. This hero is not even planned in a budget. :-(

**Comment by taurus-forever** [comment]
> My concern is that merging this PR will break the release as long as that issue is open (e.g. we might accidentally release only `focal` and not `jammy`)

Right. Feel free to remove "destructive-mode" and merge PR just to bump actions.
I will search for TypeScript developer to improve charming-action/upload-charm.
Unfortunately the code there designed in a way "one charmcraft.yaml === one charm to upload".

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -37,11 +37,12 @@ jobs:
...
           fetch-depth: 0
       - name: Select charmhub channel
-        uses: canonical/charming-actions/channel@2.1.0
+        uses: canonical/charming-actions/channel@2.2.2
         id: channel
       - name: Upload charm to charmhub
-        uses: canonical/charming-actions/upload-charm@2.1.0
+        uses: canonical/charming-actions/upload-charm@2.2.2
         with:
+          destructive-mode: false
```
Comment (taurus-forever) [inline .github/workflows/release.yaml:45]: let's remove this like to close PR and move forward with charm focal+jammy when we can.

**Review by taurus-forever — APPROVED** [review]

**Review by shayancanonical — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Nice to be up-to-date... the next stage is Renovate :-D

**Pull request merged**